src/compiler/checker.ts

```diff
@@ -10449,7 +10449,10 @@ namespace ts {
                     case SyntaxKind.FunctionExpression:
                     case SyntaxKind.FunctionDeclaration:
                     case SyntaxKind.ArrowFunction:
-                    case SyntaxKind.ObjectLiteralExpression: return false;
+                    case SyntaxKind.ClassExpression:
+                    case SyntaxKind.ClassDeclaration:
+                    case SyntaxKind.GetAccessor:
+                    case SyntaxKind.SetAccessor: return false;

```

Comment:
I also wonder whether this is correct; even if it's completely silly, I believe you can actually do something like the following:

```TypeScript
class Base {}

class A extends Base {
    constructor() {
         class C {
             [super(), Symbol.iterator]() { return undefined; } 
         }
    }
}
```

Perhaps there is a case for special casing `PropertyDeclaration` instead of using `isClassLike` by only checking whether the `name` has a `super` call.